modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/RightSidebar/RightSidebarObjectFieldDetails.tsx

```diff
@@ -94,7 +94,6 @@ export function RightSidebarObjectFieldDetails() {
 
 			delete objectField.defaultValue;
 			delete objectField.listTypeDefinitionId;
-			delete objectField.system;
```

Comment:
Deleting `objectField.system` is ok, we should keep this line as is.

The real problem is in line 88 and 92, above:
```
			if (!editedObjectField) {
				objectField = values;
			}
			else {
				objectField = editedObjectField;
			}
```
Here we are assigning states that are later modified directly, which causes unexpected results. We should assign to a copy before deleting `objectField.system`, like this:.

```
			if (!editedObjectField) {
				objectField = {...values};
			}
			else {
				objectField = {...editedObjectField};
			}
```

This way, we can delete delete `objectField.system` with no unwanted side effects.

Reply:
I see, it makes sense, I've changed what was requested, Thx @cgmonte

Reply:
@Artartur

Reply:
Yes, I did